### ISSUE TITLE
@psalm-suppress UnusedClass Not propagated to concrete classes

### ISSUE THREAD

**Issue by zmitic** [vimeo/psalm#4675]
I have hundreds of tagged services (i.e. classes that implement some interface) which are auto-wired by Symfony. On one such interface I put this:

```php
/**
 * @psalm-suppress UnusedClass
 */
interface PaymentLevelTagInterface
```

but with ``findUnusedCode="true"``, all classes are still reported as unused.

---

Shouldn't suppression on interface propagate? I have tagged services everywhere, it is not possible to configure it on directory level.

---

https://psalm.dev/r/6a86673249

**Comment by weirdan** [comment]
This sounds like something symfony plugin should do. At least that's what PHPUnit plugin does: https://github.com/psalm/psalm-plugin-phpunit/blob/a386816f6de5f4c92e955b385bc796f6a174b0a0/src/Hooks/TestCaseHandler.php#L119-L128

**Comment by zmitic** [comment]
@weirdan I was thinking about plugin, same ideas was mentioned in other issue. But isn't this something that should be on psalm level?

Rationale: interface is a contract and if contract says ``don't report unused class``, then all classes implementing that interface have to follow contract.

Is that correct?

**Comment by weirdan** [comment]
I guess I don't see suppressions as a part of contract.

**Comment by zmitic** [comment]
@weirdan You are right. Maybe it fits into ``@inheridoc`` syntax; classes would inherit docs from parent classes and interfaces, right?
